### ISSUE TITLE
Improve performance of random

### ISSUE THREAD

**Issue by JesseWeinstein** [sampsyo/beets#1795]
Currently, the plugin generates the full query, then samples it in Python. 
I suspect it would be faster to randomly sort the query in SQL, and LIMIT it to the number requested.

https://stackoverflow.com/questions/1253561/sqlite-order-by-rand

**Comment by sampsyo** [comment]
Yep, this would be a great idea. Doing so would also probably make random queries more composable, so stuff like #795 would be easier to accomplish. See also #750.
